src/modules/vtol_att_control/standard.cpp

```diff
@@ -237,7 +240,9 @@ void Standard::update_transition_state()
 
 		} else if (_pusher_throttle <= _param_vt_f_trans_thr.get()) {
 			// ramp up throttle to the target throttle value
-			_pusher_throttle = _param_vt_f_trans_thr.get() * time_since_trans_start / _param_vt_psher_rmp_dt.get();
+			_pusher_throttle = _pusher_start_trans + _param_vt_f_trans_thr.get()
```

Comment:
Can't it just be a max(min(_pusher_throttle, _param_vt_f_trans_thr), ramped_up_pusher_throttle)?

Like I did for tiltrotor: https://github.com/PX4/PX4-Autopilot/blob/7e49147bcf1473311339a45a92b5026e9640c846/src/modules/vtol_att_control/tiltrotor.cpp#L319 (The min() I didn't add there, would in theory though also make sense.

Reply:
It's true that is it quite a shame not to start ramping up from the beginning of the transition and wait idle for a while as you explained @ThomasRigi but on the other hand I don't see how your idea can be implemented without adding a state, which isn't ideal. Indeed, there is no dt already declared that would simplify the computation of the new throttle, and to define one we would need to keep in memory the last timestamp and thus create a global state.

About your proposal to change the parameter into a slope one, it is looks like a good idea indeed ! I just have one concern: the time for rap up parameter is also used in another part of the code, during the back transition, to calculate the throttle for active braking and I don't really see a way to adapt the code to the new VT_PSHER_SLEW.

Reply:
There already is a _dt in https://github.com/PX4/PX4-Autopilot/blob/main/src/modules/vtol_att_control/vtol_type.h#L222

As for the occurrence of the ramp up time for the back transition reverse thrust, I'm sure we can find an equivalent solution using the new VT_PSHER_SLEW parameter.